src/Bridges/DoctrineOrm/GenerateCommand.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/**
+ * This file is part of the Nextras community extensions of Nette Framework
+ *
+ * @license    New BSD License
+ * @link       https://github.com/nextras/migrations
+ */
+
+namespace Nextras\Migrations\Bridges\DoctrineOrm;
+
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Tools\SchemaTool;
+use Nextras;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\BufferedOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+
+
+class GenerateCommand extends Command
+{
+	/** @var EntityManager */
+	private $em;
+
+	/** @var Application */
+	private $app;
+
+	/** @var string|NULL */
+	private $ignoredQueriesFile;
+
+
+	/**
+	 * @param EntityManager $em
+	 * @param Application   $app
+	 * @param string|NULL   $ignoredQueriesFile
+	 */
+	public function __construct(EntityManager $em, Application $app, $ignoredQueriesFile)
+	{
+		parent::__construct();
+		$this->em = $em;
+		$this->app = $app;
+		$this->ignoredQueriesFile = $ignoredQueriesFile;
+	}
+
+
+	protected function configure()
+	{
+		$this->setName('migrations:generate');
+		$this->setDescription('Generates a new migration file with Doctrine SchemaTool');
+		$this->addArgument('label', InputArgument::REQUIRED, 'short description');
+	}
+
+
+	protected function execute(InputInterface $input, OutputInterface $output)
+	{
+		$ignoredQueries = $this->getIgnoredQueries();
+		$updateQueries = $this->getUpdateQueries();
+		$updateQueries = array_diff($updateQueries, $ignoredQueries);
+
+		if (count($updateQueries) === 0) {
+			$output->writeln('Nothing to update');
+			return 0;
+		}
+
+		$path = $this->createMigrationFile($input->getArgument('label'));
+		$content = implode(";\n", $updateQueries) . ";\n";
+
+		if (file_put_contents("$path.tmp", $content) !== strlen($content) || !rename("$path.tmp", $path)) {
+			$output->writeln("Unable to write to '$path'.");
+			return 1;
+		}
+
+		$output->writeln($path);
+		return 0;
+	}
+
+
+	/**
+	 * @return string[]
+	 */
+	protected function getIgnoredQueries()
+	{
+		if ($this->ignoredQueriesFile) {
+			$content = file_get_contents($this->ignoredQueriesFile);
+			$queries = array_filter(preg_split('~;(\n|\z)~', $content));
+
+		} else {
+			$queries = [];
+		}
+
+
+		return $queries;
+	}
+
+
+	/**
+	 * @return string[]
+	 */
+	protected function getUpdateQueries()
+	{
+		$schemaTool = new SchemaTool($this->em);
+		$queries = $schemaTool->getUpdateSchemaSql($this->em->getMetadataFactory()->getAllMetadata(), TRUE);
```

Comment:
this could be set in constructor, either by moving both schemaTool and metadata to state and removing em, or moving this to closure. Both remove em from state, which seems desirable as it's quite a weird dependency.